tools/syz-trace2syz/proggen/vma_call_handlers.go

```diff
@@ -0,0 +1,349 @@
+package proggen
+
+import (
+	"github.com/google/syzkaller/pkg/log"
+	"github.com/google/syzkaller/prog"
+	"github.com/google/syzkaller/tools/syz-trace2syz/parser"
+)
+
+func parseMemoryCall(ctx *Context) *prog.Call {
+	syzCall := ctx.CurrentSyzCall
+	straceCall := ctx.CurrentStraceCall
+
+	switch straceCall.CallName {
+	case "mmap":
+		return parseMmap(syzCall.Meta, straceCall, ctx)
+	case "mremap":
+		return parseMremap(syzCall.Meta, straceCall, ctx)
+	case "msync":
+		return parseMsync(syzCall.Meta, straceCall, ctx)
+	case "mprotect":
+		return parseMprotect(syzCall.Meta, straceCall, ctx)
+	case "munmap":
+		return parseMunmap(syzCall.Meta, straceCall, ctx)
+	case "madvise":
+		return parseMadvise(syzCall.Meta, straceCall, ctx)
+	case "mlock":
+		return parseMlock(syzCall.Meta, straceCall, ctx)
+	case "munlock":
+		return parseMunlock(syzCall.Meta, straceCall, ctx)
+	case "shmat":
+		return parseShmat(syzCall.Meta, straceCall, ctx)
+	}
+	return nil
+}
+
+func parseMmap(mmap *prog.Syscall, syscall *parser.Syscall, ctx *Context) *prog.Call {
+	call := &prog.Call{
+		Meta: mmap,
+		Ret:  prog.MakeReturnArg(mmap.Ret),
+	}
+	ctx.CurrentSyzCall = call
+
+	length := (parseLength(syscall.Args[1], ctx)/ctx.Target.PageSize + 1) * ctx.Target.PageSize //RoundUp PageSize
```

Comment:
This looks wrong. If the original value is multiple of PageSize, then the result will be 1 page larger.
I think we want: `(x + PageSize - 1) / PageSize * PageSize`